etna/articles/models.py

```diff
@@ -269,48 +271,67 @@ def similar_items(self) -> Tuple["ArticlePage"]:
             return ()
 
         # Identify 'other' live pages with tags in common
-        tag_match_ids = (
-            ArticlePage.objects.public()
-            .live()
-            .not_page(self)
-            .filter(tagged_items__tag_id__in=tag_ids)
-            .values_list("id", flat=True)
-            .distinct()
-        )
-        if not tag_match_ids:
+        matching_page_ids = []
+
+        for page_type in [ArticlePage, FocusedArticlePage, RecordArticlePage]:
+            matching_page_ids.extend(
+                page_type.objects.live()
+                .public()
+                .not_page(self)
+                .filter(tagged_items__tag_id__in=tag_ids)
+                .values_list("id", flat=True)
+                .distinct()
+            )
```

Comment:
I don't think you need the three separate queries here, as all you're after is page ids (which can be fetched in one go):
```suggestion
        matching_page_ids = (
            Page.objects.live()
            .public()
            .not_page(self)
            .exact_type(ArticlePage, FocusedArticlePage, RecordArticlePage)
            .filter(tagged_items__tag_id__in=tag_ids)
            .values_list("id", flat=True)
        )
    
           
```